src/Evolution/Systems/RadiationTransport/M1Grey/M1Closure.cpp

```diff
@@ -0,0 +1,284 @@
+          2. * (h_0_v * h_thick_v * v_sqr_pt + h_0_f * h_thick_f * s_sqr_pt +
+                h_0_v * h_thick_f * v_dot_f_pt +
+                h_0_f * h_thick_v * v_dot_f_pt - h_0_n * h_thick_n);
+      const double h_sqr_thin_thick =
+          2. *
+          (h_thin_v * h_thick_v * v_sqr_pt + h_thin_f * h_thick_f * s_sqr_pt +
+           h_thin_v * h_thick_f * v_dot_f_pt +
+           h_thin_f * h_thick_v * v_dot_f_pt - h_thin_n * h_thick_n);
+      const double h_sqr_thick_thick =
+          square(h_thick_v) * v_sqr_pt + square(h_thick_f) * s_sqr_pt +
+          2. * h_thick_v * h_thick_f * v_dot_f_pt - square(h_thick_n);
+      const double h_sqr_thin_thin =
+          square(h_thin_v) * v_sqr_pt + square(h_thin_f) * s_sqr_pt +
+          2. * h_thin_v * h_thin_f * v_dot_f_pt - square(h_thin_n);
+
+      // Root finding function
+      const auto zeta_j_sqr_minus_h_sqr = [
+        &e_pt, &j_0, &j_thin, &j_thick, &h_sqr_0, &h_sqr_thick, &h_sqr_thin,
+        &h_sqr_thin_thin, &h_sqr_thick_thick, &h_sqr_thin_thick
+      ](double local_zeta) noexcept {
+        const double chi = minerbo_closure_function(local_zeta);
+        const double dchi_dzeta = minerbo_closure_deriv(local_zeta);
+        const double d_thin = 1.5 * chi - 0.5;
+        const double d_thick = 1. - d_thin;
+        const double d_thin_dzeta = 1.5 * dchi_dzeta;
+        const double d_thick_dzeta = -d_thin_dzeta;
+
+        const double e_fluid = j_0 + j_thin * d_thin + j_thick * d_thick;
+        const double de_fluid_dzeta =
+            j_thin * d_thin_dzeta + j_thick * d_thick_dzeta;
+        const double h_sqr = h_sqr_0 + h_sqr_thick * d_thick +
+                             h_sqr_thin * d_thin +
+                             h_sqr_thin_thin * square(d_thin) +
+                             h_sqr_thick_thick * square(d_thick) +
+                             h_sqr_thin_thick * d_thin * d_thick;
+        const double dh_sqr_dd_thin = h_sqr_thin + h_sqr_thin_thick * d_thick +
+                                      2. * h_sqr_thin_thin * d_thin;
+        const double dh_sqr_dd_thick = h_sqr_thick + h_sqr_thin_thick * d_thin +
+                                       2. * h_sqr_thick_thick * d_thick;
+        return std::make_pair(
+            (square(e_fluid * local_zeta) - h_sqr) / square(e_pt),
+            (2. * e_fluid * de_fluid_dzeta * square(local_zeta) +
+             2. * square(e_fluid) * local_zeta - d_thin_dzeta * dh_sqr_dd_thin -
+             d_thin_dzeta * dh_sqr_dd_thick) /
+                square(e_pt));
+      };
+      const double& zeta = get(*closure_factor)[s];
+      // Initial guess for root finding.
+      // Choice 1: previous value
+      // Choice 2: value for zero-velocity
+      const double zeta_guess =
+          (zeta <= avoid_divisions_by_zero or zeta >= 1. ? sqrt(s_sqr_pt) / e_pt
+                                                         : zeta);
+      get(*closure_factor)[s] = RootFinder::newton_raphson(
```

Comment:
add a variable at the beginning of the function defining `constexpr size_t root_find_number_of_digits = 6` and use it here (and document it in the doxygen comment)